src/Symfony/Component/Security/Core/Authentication/RememberMe/PersistentTokenInterface.php

```diff
@@ -37,7 +37,7 @@ public function getTokenValue(): string;
     /**
      * Returns the time the token was last used.
      */
-    public function getLastUsed(): \DateTime;
+    public function getLastUsed(): \DateTimeInterface;
```

Comment:
This is a BC breaks for consumers of the interface

Reply:
IIUC banning DateTime, does not meaning banning DateTimeInterface?

Reply:
This is a BC break for the caller, as they could rely on the method returning `DateTime` before. E.g. this code works before this change, but breaks after:

```php
$lastUsed = $token->getLastUsed();
$lastUsed->modify('+30 minutes');

if ($lastUsed < new \DateTimeImmutable('now')) {
    throw new \RuntimeException('This token was last used 30 minutes ago');
}
```

Reply:
The consumer of a return type is not choosing whether it uses DateTime or a DateTimeInterface. It has to handle everything.

Reply:
Also next major versions cannot have BC breaks that didn't have a deprecation (and way to get out of deprecation) in the last minor before it.

We would have to create a new method with the new return type and deprecate the old method. But I'm personally not sure if that is worth is: What about delaying to remove mutable DateTime in user-facing code?

Reply:
@nicolas-grekas I ping you here. I think you should be part of this discussion

Reply:
I've no perfect solution to this. We would document a way to detect all such uses of `DateTime` and rewrite them automatically:

Before:
```php
$lastUsed = $token->getLastUsed();
$lastUsed->modify('+30 minutes');
```
After:
```php
$lastUsed = $token->getLastUsed();
$lastUsed = $lastUsed->modify('+30 minutes');
```

That'd make the code the same whether `DateTime` or `DateTimeImmutable` is used.
Can Rector do that? /cc @TomasVotruba 
Should the PHP engine trigger a deprecation when the return value of one of DateTime's setters is not used? I'm not sure how this would be possible technically, but let's just dream a bit :) /cc @derickr

Reply:
@nicolas-grekas If the type is known, Rector can handle it :)

Reply:
And regarding the widened type, we might need something like `ReturnTypeWillChange` to indicate that this interface will change its return type in the future. But there is no userland solution for that.